### PR TITLE
Post picker decode title

### PR DESCRIPTION
**Pull request by juliobranha**
### Summary by CodeRabbit

- Enhancement: Improved the display of post titles in the post picker tool. Special characters in titles are now correctly decoded, ensuring they are displayed as intended. This change enhances the user experience by providing accurate and readable post titles.
- Maintenance: Updated several components within the post picker tool to import the `JSX` type from the 'react' package. This is a behind-the-scenes update that maintains the tool's code quality and does not affect the user experience.

### CODE REVIEW

File: packages/block-editor-tools/src/components/post-picker/post.tsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { sprintf } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+
```
Comment (juliobranha) [inline packages/block-editor-tools/src/components/post-picker/post.tsx:3]:
```suggestion
```

File: packages/block-editor-tools/package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
   "homepage": "https://github.com/alleyinteractive/alley-scripts#readme",
   "dependencies": {
     "@alleyinteractive/tsconfig": "*",
+    "@storybook/react": "^7.5.1",
```
Comment (juliobranha) [inline packages/block-editor-tools/package.json:48]: Needed to address ts errors

Comment (dlh01) [reply packages/block-editor-tools/package.json:48]: This is needed in the published package dependencies, not the dev dependencies? That seems peculiar.

File: packages/block-editor-tools/src/components/post-picker/post.tsx [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { sprintf } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
```
Comment (dlh01) [inline packages/block-editor-tools/src/components/post-picker/post.tsx:2]: Alpha order?

**Review by dlh01 — APPROVED** [review]
The changes to the post picker seem fine, but I would seek a second opinion about what's needed to satisfy TypeScript. I don't trust my own judgment about it.

File: packages/block-editor-tools/package.json (hunk 2 above)
Comment (juliobranha) [reply packages/block-editor-tools/package.json:48]: Yeah, not sure I understand it 100% either, but `npm run build` wouldn't work without that

Comment (juliobranha) [reply packages/block-editor-tools/package.json]: Actually moved to dev dependencies and is fine, thanks!

**Pull request merged**